README.md

```diff
@@ -1,78 +1,77 @@
 
-This will calculate major and minor prefix tags for when you want to provide users with a way to obtain the latest `major` or `major.minor` version.
+This action can be useful if a git tag like `1.2.3` is pushed to a GitHub repo and you want a GitHub Workflow's job to not only take action with regards to the pushed tag `1.2.3` specifically, but also to related tags like `1.2`, `1`, and `latest` assuming the pushed tag's version is new enough to represent those more general tags.
 
-Suffixes of the form `-[0-9]+` are supported since this is often used as a build number.
-Any other suffixes are treated as a pre-release, and are not supported.
+## Example output
 
-If this called on a branch instead of a tag just the branch-name is returned as the tag.
+The GitHub action's only output is named `tags` and is a JSON formatted list. See the example workflow below for details on how to convert the JSON formatted list to something else.
 
-## Examples
-
-Current and latest tag: `1.2.3`, returned tags: `[1.2.3, 1.2, 1, latest]`.
-
-Current tag `1.2.3` but the repository already contains a more recent tag `2.0.0`: return `[1.2.3, 1.2, 1]`.
-
-Branch `main`, return `[main]`.
+| Pushed reference | GitHub repo tags | `tags` output                        | Comment                                                                                |
+| ---------------- | ---------------- | ------------------------------------ | -------------------------------------------------------------------------------------- |
+| ``               | ...              | `"[]"`                               |
+| `main`           | ...              | `"[main]"`                           | Branches                                                                               |
+| `1.2.3`          | `1.2.0`          | `"[1.2.3, 1.2, 1, latest]"`          |
+| `1.2.3`          | `1.2.0`, `2.0.0` | `"[1.2.3, 1.2, 1]"`                  |
+| `1.2.3`          | `1.2.0`, `1.3.0` | `"[1.2.3, 1.2]"`                     |
+| `1.2.3`          | `1.2.0`, `1.2.5` | `"[1.2.3]"`                          |
+| `1.2.3-alpha.1`  | `1.2.0`          | `"[1.2.3-alpha.1]"`                  | Non-supported pre-release tags                                                         |
+| `1.2.3-4`        | `1.2.0`, `1.2.3` | `"[1.2.3-4, 1.2.3, 1.2, 1, latest]"` | Supported pre-release tags (build numbers, with a version suffix pattern of `-[0-9]+`) |
 
 ## Required input parameters
 
 - `githubToken`: The GitHub token, required so this action can fetch tags using the GitHub API.
 
-## Optional parameters
+## Optional input parameters
 
 - `prefix`: A string that each returned tag should be prefixed with, for example to tag a Docker container set this to `user/repository:`.
 
-## Example
+## Example workflow
 
 ```yaml
-name: Docker build
+name: Build and publish image to DockerHub
 
 on:
   push:
```

Comment:
```suggestion
  push:
    # Uncomment this if you also want a tag created for all pushed branches
    # branches:
    #   - "*"
```
What do you reckon to this?

Reply:
Ah! It was opt-out, i omitted it entirely, and now you suggest opt-in - absolutely! I was a bit opinionated about opt-out as I found it to not serve a purpose to have it in our example.